framework/src/mediaplayer/mediaplayer.c

```diff
@@ -0,0 +1,552 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/audio/audio.h>
+#include <tinyara/audio/pcm.h>
+#include <tinyalsa/tinyalsa.h>
+#include <tinyara/fs/ioctl.h>
+#include <pthread.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sched.h>
+#include <string.h>
+#include <mediaplayer/mediaplayer.h>
+#include "mplayer_debug.h"
+
+/***
+ * Structure
+ */
+struct media_s {
```

Comment:
Why do you define this in c file? To use mediaplayer public APIs, this is needed.

Reply:
The header file includes the reference to media_s structure. I did not want to open the media_s internal structure to application , so i defined the struct in c file.